go/vt/vtgate/planbuilder/abstract/operator.go

```diff
@@ -29,12 +29,16 @@ type (
 	//     while still preserving the results
 	//	*  LeftJoin - A left join. These can't be evaluated in any order, so we keep them separate
 	//	*  Join - A join represents inner join.
+	//  *  SubQuery - A represents a query that encapsulates one or more sub-queries (SubQueryInner).
 	Operator interface {
 		// TableID returns a TableSet of the tables contained within
 		TableID() semantics.TableSet
 
 		// PushPredicate pushes a predicate to the closest possible operator
 		PushPredicate(expr sqlparser.Expr, semTable *semantics.SemTable) error
+
+		// UnsolvedPredicates returns any predicates that have dependencies from the outside
```

Comment:
this comment could be worked on. not easy to understand what it means

Reply:
I changed it in ea9871c3f1, to:

```
// UnsolvedPredicates returns any predicates that have dependencies on the given Operator and
// on the outside of it (a parent Select expression, any other table not used by Operator, etc).
```